torch/_dynamo/utils.py

```diff
@@ -404,6 +405,8 @@ def clone_input(x):
                 result.requires_grad_(x.requires_grad)
             if x.is_leaf and x.grad is not None:
                 result.grad = clone_input(x.grad)
+            if is_param:
+                result = torch.nn.Parameter(result)
         except RuntimeError:
```

Comment:
unrelated: it would be great to remove the runtime error path and make this always succeed